### PR TITLE
Remove data validation from item populating

### PR DESCRIPTION
**Pull request by hannunosto**

### CODE REVIEW

File: app/code/community/Nosto/Tagging/Model/Meta/LineItem.php [hunk 1]
```diff
@@ -87,11 +78,11 @@ public function __construct(array $args)
             throw new InvalidArgumentException(sprintf('%s.currency must be an instance of NostoCurrencyCode.', __CLASS__));
         }
 
-        $this->_productId = $args['productId'];
-        $this->_quantity = $args['quantity'];
-        $this->_name = $args['name'];
-        $this->_unitPrice = $args['unitPrice'];
-        $this->_currency = $args['currency'];
+        $this->_productId = isset($args['productId']) ? $args['productId'] : false;
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/LineItem.php]: When setting to false, does it send `false` in the JSON or an empty string?

File: app/code/community/Nosto/Tagging/Model/Meta/LineItem.php [hunk 2]
```diff
@@ -71,15 +71,6 @@
...
-            throw new InvalidArgumentException(sprintf('%s.productId must be a integer value.', __CLASS__));
-        }
-        if (!isset($args['quantity']) || !is_int($args['quantity']) || !($args['quantity'] > 0)) {
-            throw new InvalidArgumentException(sprintf('%s.quantity must be a integer value above zero.', __CLASS__));
-        }
-        if (!isset($args['name']) || !is_string($args['name']) || empty($args['name'])) {
-            throw new InvalidArgumentException(sprintf('%s.name must be a non-empty string value.', __CLASS__));
-        }
         if (!($args['unitPrice'] instanceof NostoPrice)) {
             throw new InvalidArgumentException(sprintf('%s.unitPrice must be an instance of NostoPrice.', __CLASS__));
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/LineItem.php]: Should these exceptions be removed as well? We can handle it in the order object.

File: app/code/community/Nosto/Tagging/Model/Meta/Order.php [hunk 3]
```diff
@@ -186,8 +186,18 @@ protected function buildItem(Mage_Sales_Model_Order_Item $item, Mage_Sales_Model
     {
         $priceHelper = Mage::helper('nosto_tagging/price');
         $store = $order->getStore();
-        $convertedPrice = $priceHelper->convertToDefaultCurrency($item->getPriceInclTax(), $store);
-        $defaultCurrencyCode = $store->getDefaultCurrency()->getCode();
+
+        try {
+            $nostoCurrencyCode = new NostoCurrencyCode($store->getDefaultCurrencyCode());
+        } catch (NostoInvalidArgumentException $e) {
+            $nostoCurrencyCode = new NostoCurrencyCode('XXX'); // fallback to no currency
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/Order.php]: Log the error as a warning.

File: app/code/community/Nosto/Tagging/Model/Meta/Order.php [hunk 4]
```diff
@@ -186,8 +186,18 @@ protected function buildItem(Mage_Sales_Model_Order_Item $item, Mage_Sales_Model
...
+        }
+
+        try {
+            $nostoPrice = new NostoPrice($priceHelper->convertToDefaultCurrency($item->getPriceInclTax(), $store));
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/Order.php]: Log the error as a warning.

File: app/code/community/Nosto/Tagging/Model/Meta/Order.php [hunk 5]
```diff
@@ -186,8 +186,18 @@ protected function buildItem(Mage_Sales_Model_Order_Item $item, Mage_Sales_Model
...
+        } catch (NostoInvalidArgumentException $e) {
+            $nostoPrice = new NostoPrice(0);
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/Order.php]: Log the error as a warning with the invalid currency code.

**Comment by mridang** [comment]
There is more validation in the `Buyer` and `Status` classes that should be handled as well.

**Comment by hannunosto** [comment]
Fix responses to be empty string & added logging. Remove validation from Buyer and Status classes

File: app/code/community/Nosto/Tagging/Model/Meta/Order/Status.php [hunk 6]
```diff
@@ -62,23 +62,9 @@ class Nosto_Tagging_Model_Meta_Order_Status extends Mage_Core_Model_Abstract imp
      */
     public function __construct(array $args)
     {
-        if (!isset($args['code']) || !is_string($args['code']) || empty($args['code'])) {
-            throw new InvalidArgumentException(sprintf('%s.code must be a non-empty string value.', __CLASS__));
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/Order/Status.php]: Should we really remove the `is_string` check?

File: app/code/community/Nosto/Tagging/Model/Meta/Order/Status.php [hunk 7]
```diff
@@ -62,23 +62,9 @@ class Nosto_Tagging_Model_Meta_Order_Status extends Mage_Core_Model_Abstract imp
...
-                throw new InvalidArgumentException(sprintf('%s.createdAt must be an instance of NostoDate.', __CLASS__));
-            }
-        }
-
-        $this->_code = $args['code'];
-        $this->_label = !empty($args['label']) ? $args['label'] : $args['code'];
-		$this->_createdAt = isset($args['createdAt']) ? $args['createdAt'] : null;
+        $this->_code = !empty($args['code']) ? $args['code'] : '';
+        $this->_label = !empty($args['label']) ? $args['label'] : '';
+		$this->_createdAt = isset($args['createdAt']) ? $args['createdAt'] : '';
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/Order/Status.php]: Fix indentation. Set `createdAt` to null

File: app/code/community/Nosto/Tagging/Model/Meta/Order.php [hunk 8]
```diff
@@ -186,8 +186,35 @@ protected function buildItem(Mage_Sales_Model_Order_Item $item, Mage_Sales_Model
...
+        try {
+            $nostoPrice = new NostoPrice($priceHelper->convertToDefaultCurrency($item->getPriceInclTax(), $store));
+        } catch (NostoInvalidArgumentException $e) {
+            Mage::log(
+                sprintf(
+                    'No default price found for item %s in store %s',
+                    $item->getId(),
+                    $store->getId()
+                ),
+                'nostotagging.log'
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/Order.php]: What's the logging level?

File: app/code/community/Nosto/Tagging/Model/Meta/Order.php [hunk 9]
```diff
@@ -186,8 +186,35 @@ protected function buildItem(Mage_Sales_Model_Order_Item $item, Mage_Sales_Model
...
+            Mage::log(
+                sprintf(
+                    'No default display or invalid currency code (%s) found for store %s',
+                    $store->getDefaultCurrencyCode(),
+                    $store->getId()
+                ),
+                Zend_Log::WARN,
+                'nostotagging.log'
+            );
+            $nostoCurrencyCode = new NostoCurrencyCode('XXX'); // fallback to no currency
```
Comment (mridang) [inline app/code/community/Nosto/Tagging/Model/Meta/Order.php]: Remove th defaults.

**Comment by hannunosto** [comment]
Outdated

**Pull request closed without merging**